### PR TITLE
TST: manual workflow to install latest RC from test pypi

### PR DESCRIPTION
**Pull request by jklenzing**
# Description

Adds a new workflow to  install the RC through the test pypi server for each major operating system.  Designed to catch any issues that may stem from improper local build and upload.

## Type of change

- New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Github Actions

# Checklist:

- [x] Make sure you are merging into the ``develop`` (not ``main``) branch
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes
- [x] Update zenodo.json file for new code contributors

### CODE REVIEW

File: .github/workflows/pip_rc_install.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# This workflow will install Python dependencies, run tests and lint with a variety of Python versions
```
Comment (jklenzing) [inline .github/workflows/pip_rc_install.yml:1]:
```suggestion
# This workflow will install Python dependencies and the latest RC of pysatNASA from test pypi.
# This test should be manually run before a pysatNASA RC is officially approved and versioned.
```

**Comment by jklenzing** [comment]
Pinging @rstoneback @aburrell @JonathonMSmith.  This should close one of the gaps in our RC testing before releases, and should probably be deployed across the ecosystem.  I'm not sure if we should have any other tests here, or if installing from test pypi is sufficient.

File: .github/workflows/pip_rc_install.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+on: [workflow_dispatch]
+
+jobs:
+  build:
+    strategy:
+      fail-fast: false
+      matrix:
+        os: ["ubuntu-latest", "macos-latest", "windows-latest"]
+        python-version: ["3.10"]
```
[the remark this answers is not in the document]
Comment (jklenzing) [reply .github/workflows/pip_rc_install.yml]: Updated at https://github.com/pysat/pysatNASA/wiki/Release-Candidate-Instructions

**Comment by jklenzing** [comment]
I've updated comments in all the workflow files accordingly.

**Review by aburrell — APPROVED** [review]

**Pull request merged**